pkg/clusterapi/controlplane.go

```diff
@@ -104,39 +90,3 @@ func (cp *ControlPlane[C, M]) UpdateImmutableObjectNames(
 
 	return nil
 }
-
-// EnsureNewNameIfChanged updates an object's name if such object is different from its current state in the cluster
-func EnsureNewNameIfChanged[M Object](ctx context.Context,
```

Comment:
This hasn't changed at all, I'm just moving the code to a different file
Same with the tests